controllers/actions.github.com/ephemeralrunnerset_controller.go

```diff
@@ -468,6 +468,37 @@ func (r *EphemeralRunnerSetReconciler) actionsClientFor(ctx context.Context, rs
 		opts = append(opts, actions.WithProxy(proxyFunc))
 	}
 
+	tlsConfig := rs.Spec.GitHubServerTLS
+	if tlsConfig != nil && tlsConfig.RootCAsConfigMapRef != "" {
+		var rootCAsConfigMap corev1.ConfigMap
+		err := r.Get(
+			ctx,
+			types.NamespacedName{
+				Namespace: rs.Namespace,
+				Name:      tlsConfig.RootCAsConfigMapRef,
+			},
+			&rootCAsConfigMap,
+		)
+		if err != nil {
+			return nil, fmt.Errorf(
+				"failed to get configmap %s: %w",
+				tlsConfig.RootCAsConfigMapRef,
+				err,
+			)
+		}
+
+		certs, err := actions.RootCAsFromConfigMap(rootCAsConfigMap.Data)
+		if err != nil {
+			return nil, fmt.Errorf(
+				"failed to get certificates from configmap %s: %w",
+				tlsConfig.RootCAsConfigMapRef,
+				err,
+			)
+		}
+
```

Comment:
can we trace out that we are creating the client with a CACert?
does it make sense to do similar things to the proxy, and trace out that the client is using a proxy?
you can do a follow-up PR if you think the PR is getting bigger and bigger. 😄

Reply:
same as https://github.com/actions/actions-runner-controller/pull/2268#discussion_r1123112719